pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -206,6 +207,15 @@ func (c *CheckConfig) GetStaticTags() []string {
 	if c.IPAddress != "" {
 		tags = append(tags, deviceIPTagKey+":"+c.IPAddress)
 	}
+
+	if c.UseDeviceIDAsHostname {
+		hostname, err := coreutil.GetHostname(context.TODO())
```

Comment:
**❓ question** 
Shouldn't we use `context.Background()` instead?‏

Reply:
@gmarabout Technically, they both return an empty context and there is no different. There are semantically different as discussed here: 

context.TODO():
> TODO returns a non-nil, empty Context. Code should use context.TODO when it’s unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

context.Background():
> Background returns a non-nil, empty Context. It is never canceled, has no values, and has no deadline. It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests.

In the code base, people use mostly TODO():
![image](https://user-images.githubusercontent.com/49917914/147209097-93b9f8dd-c887-4872-b9f6-b9705e8de102.png)

In our case, I think TODO() is fine since we might want to replace with some other contexts if there is something more suitable later.